kotlin/spatial-operations/src/main/java/com/esri/arcgisruntime/sample/spatialoperations/MainActivity.kt

```diff
@@ -0,0 +1,217 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.spatialoperations
+
+import android.graphics.Color
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.geometry.Geometry
+import com.esri.arcgisruntime.geometry.GeometryEngine
+import com.esri.arcgisruntime.geometry.Part
+import com.esri.arcgisruntime.geometry.PartCollection
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.PointCollection
+import com.esri.arcgisruntime.geometry.Polygon
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  private val inputGeometryOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+  private val resultGeometryOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+
+  // simple black (0xFF000000) line symbol for outlines
+  private val lineSymbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLACK, 1f)
+  private val resultFillSymbol = SimpleFillSymbol(
+    SimpleFillSymbol.Style.SOLID, -0x16e0e1, lineSymbol
+  )
+  private var inputPolygon1: Polygon? = null
+  private var inputPolygon2: Polygon? = null
+
+  // The spatial operation switching menu items.
+  private var noOperationMenuItem: MenuItem? = null
+  private var intersectionMenuItem: MenuItem? = null
+  private var unionMenuItem: MenuItem? = null
+  private var differenceMenuItem: MenuItem? = null
+  private var symmetricDifferenceMenuItem: MenuItem? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    mapView.apply {
+      // create ArcGISMap with topographic basemap
+      map = ArcGISMap(Basemap.createLightGrayCanvas())
+
+      // create graphics overlays to show the inputs and results of the spatial operation
+      graphicsOverlays.add(inputGeometryOverlay)
+      graphicsOverlays.add(resultGeometryOverlay)
+
+      // create input polygons and add graphics to display these polygons in an overlay
+      createPolygons()
```

Comment:
I'm still super new to Kotlin, but it's my understanding that when using the `apply` scoping function, we shouldn't be adding methods whose logic doesn't relate directly to the object using apply. In this instance `createPolygons` is creating polygons, which isn't directly linked to the set up of the map view object. I'm wondering would that be better called outside of the `mapView.apply` block, along with the following envelope and `setViewpointGeometryAsync` set up and am raising this as a discussion point. What do you think @TADraeseke?

Reply:
yes I am a bit shaky on what can and cannot be done in apply{}. It seems it is intended for initialization, but I am a bit lost with the merits of calling something like createPolygons() inside of apply{} versus the merits of, say, chaining a run {} onto the end of it, or just adding a few more lines of code. Most advice I find is to use the scope functions to reduce the amount of code needed, so that is why I called createPolygons() in apply {}. At first I was worried that the newly constructed object might not be fully intialized in apply{}, but this seems to not be the case (can someone confirm this?) and that the choice of scope function is a question of convention or design pattern.